--- h5_interm_group.c	2024-11-13 11:56:00.913059201 -0600
+++ h5_interm_group_logvol.c	2024-11-13 11:56:54.281932771 -0600
@@ -84,11 +84,12 @@
         g2_id  = H5Gopen2(file, "/G1/G2", H5P_DEFAULT);
         status = H5Gget_info(g2_id, &g2_info);
         printf("Group /G1/G2 has %d member(s)\n", (int)g2_info.nlinks);
-
+#if 0
         for (i = 0; i < (int)g2_info.nlinks; i++) {
             H5Lget_name_by_idx(g2_id, ".", H5_INDEX_NAME, H5_ITER_NATIVE, (hsize_t)i, name, 3, H5P_DEFAULT);
             printf("Object's name is %s\n", name);
         }
+#endif
         H5Gclose(g2_id);
     }
     H5Fclose(file);
